serverless/internal/storage/webstorage/fs.go

```diff
@@ -0,0 +1,290 @@
+	// if Sequence has been called since the last time an Integrate/WriteCheckpoint
+	// was called.
+	for {
+		seq := fs.nextSeq
+
+		// Ensure the sequencing directory structure is present:
+		seqDir, seqFile := layout.SeqPath(fs.root, seq)
+
+		// Write the newly sequenced leaf
+		seqPath := filepath.Join(seqDir, seqFile)
+		err := createExclusive(seqPath, leaf)
+		if err != nil {
+			if !errors.Is(err, os.ErrExist) {
+				return 0, fmt.Errorf("failed to store sequenced leaf: %v", err)
+			}
+			fs.nextSeq++
+			continue
+		}
+
+		// Create a leafhash file containing the assigned sequence number.
+		// This isn't infallible though, if we crash after hardlinking the
+		// sequence file above, but before doing this a resubmission of the
+		// same leafhash would be permitted.
+		if err := createExclusive(leafFQ, []byte(strconv.FormatUint(seq, 16))); err != nil {
+			return 0, fmt.Errorf("couldn't create temporary leafhash file: %w", err)
+		}
+		// All done!
+		return seq, nil
+	}
+}
+
+// ScanSequenced calls the provided function once for each contiguous entry
+// in storage starting at begin.
+// The scan will abort if the function returns an error, otherwise it will
+// return the number of sequenced entries.
+func (fs *Storage) ScanSequenced(begin uint64, f func(seq uint64, entry []byte) error) (uint64, error) {
+	end := begin
+	for {
+		sp := filepath.Join(layout.SeqPath(fs.root, end))
+		entry, err := get(sp)
+		if errors.Is(err, os.ErrNotExist) {
+			// we're done.
+			return end - begin, nil
+		} else if err != nil {
+			return end - begin, fmt.Errorf("failed to read leafdata at index %d: %w", begin, err)
+		}
+		if err := f(end, entry); err != nil {
+			return end - begin, err
+		}
+		end++
+	}
+}
+
+// GetTile returns the tile at the given tile-level and tile-index.
+// If no complete tile exists at that location, it will attempt to find a
+// partial tile for the given tree size at that location.
+func (fs *Storage) GetTile(level, index, logSize uint64) (*api.Tile, error) {
+	tileSize := layout.PartialTileSize(level, index, logSize)
+	p := filepath.Join(layout.TilePath(fs.root, level, index, tileSize))
+	t, err := get(p)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return nil, fmt.Errorf("failed to read tile at %q: %w", p, err)
+		}
+		return nil, err
+	}
+
+	var tile api.Tile
+	if err := tile.UnmarshalText(t); err != nil {
+		return nil, fmt.Errorf("failed to parse tile: %w", err)
+	}
+	return &tile, nil
+}
+
+// StoreTile writes a tile out to disk.
+// Fully populated tiles are stored at the path corresponding to the level &
+// index parameters, partially populated (i.e. right-hand edge) tiles are
+// stored with a .xx suffix where xx is the number of "tile leaves" in hex.
+func (fs *Storage) StoreTile(level, index uint64, tile *api.Tile) error {
+	tileSize := uint64(tile.NumLeaves)
+	glog.V(2).Infof("StoreTile: level %d index %x ts: %x", level, index, tileSize)
```

Comment:
This is the only instance of `glog` in here. Would not using it allow a smaller wasm to be generated (faster) ?

If there was other critical logging then I'd not suggest this, but I don't see this logging ever being used as its V(2), but that can only be determined at runtime.

Reply:
Only saves ~700 bytes in the complied file, but you're right to call it out - removed it.